app/controllers/avo/associations_controller.rb

```diff
@@ -42,19 +44,24 @@ def show
 
     def new
       @resource.hydrate(record: @record)
+    end
 
+    def options
```

Comment:
To increase performance we can serialize and encrypt the query and pass from new to options through params avoiding the query compute on each options request.